### ISSUE TITLE
Could line execution count be added?

### ISSUE THREAD

**Issue by pjlewis** [jacoco/jacoco#89]
Having switched from Cobertura I have started to miss the count of how many times a given line has been executed during a run of my unit tests.  Could this be included in the JaCoCo HTML reports?

I found this useful for various reasons:
- Balancing out 'unbalanced' tests (where for example, one path through a method was followed hundreds of times and another path only once).
- Thinking about, and then extending, the set of tests for a given method/class.
- In particular, finding tests that should have been run passing null and non-null parameters.  A line execution count of 1 was a sure sign this had not been done [or, where n parameters could be null, either _count < n+1_ or _count < 2^n_]).  Note: A lot of my methods defer to utility methods that handle null, so this is not a branch in the tested method, just a single forwarding call.  Checking parameters and throwing exceptions are similarly handled in my code (i.e. not as branches in the tested method, but as [potentially] exception throwing utilities).
- Also, when testing similar methods - where a similar set of tests might be expected - mismatching numbers have been a good hint at incomplete testing.

Perhaps worth saying that I am working on collection classes/utilities and similar code.  Also, that I often run only the tests for a given class (or package) and review the specific coverage given by those tests (ignoring incidental coverage resulting from other tested code that happens to be implemented in terms of these other methods).

ASIDE:

It would be nice to report the distinction between 'targeted' versus 'incidental' coverage, somehow, when testing and reviewing coverage results.  For example, separating the extent that Foo_Test covers the class Foo, from the extent that _all_ unit tests hit code in Foo.  I currently do this by selectively running Foo_Test when working on the Foo class, and checking the coverage.  This is okay in development but not much use when reviewing the entire code base.

To give a concrete example, I have a JDBC wrapper/helper class used for all database access - it eliminates a lot of the boilerplate from JDBC code.  It should be tested to ~100% coverage by its own tests.  However it is also extensively used, indirectly, by the CRUD testing in our system.  These CRUD tests account for most of the calls to this class - but they very rarely test fringe and degenerate cases.  I recently noticed for instance that a new method was added to this class (copy, paste and tweak).  Since this was called by one of the tested CRUD implementation, it was only a single line of yellow that flagged up the missing 'targeted' test for this class.

In summary:
- Would it be possible to show line execution count in the reports?
- Would it (some time in the future) be possible to show different statistics for 'targeted' versus 'incidental' coverage - perhaps using some sort of pattern matching to determine which are the 'targeted' tests?
